src/controller.py

```diff
@@ -0,0 +1,790 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""nginx-ingress-integrator k8s resource controllers."""
+
+import abc
+import functools
+import logging
+import time
+import typing
+
+import kubernetes.client
+
+from exceptions import InvalidIngressError
+from ingress_definition import IngressDefinition
+
+logger = logging.getLogger(__name__)
+
+AnyResource = typing.TypeVar(  # pylint: disable=invalid-name
```

Comment:
What is pylint expecting here please? Any linting disable should include a comment with the reason why

Reply:
Actually this is not required, don't know why I added this here, removed, thanks!